src/Entities/Player.cpp

```diff
@@ -89,6 +89,12 @@ cPlayer::cPlayer(cClientHandlePtr a_Client, const AString & a_PlayerName) :
 	m_UUID((a_Client != nullptr) ? a_Client->GetUUID() : ""),
 	m_CustomName("")
 {
+	if (a_PlayerName.empty() || a_PlayerName.length() > 16)
+	{
+		LOGERROR("Playername of %s is too long(>16) or empty", a_PlayerName.c_str());
+		VERIFY(false);
+	}
```

Comment:
This is the worst construct you could have used.
We use `LOGERROR` only for stuff that the server cannot recover from and terminates. A long username should definitely NOT be such a case.
And `VERIFY(false)` is the stupidest construct ever. You're basically saying "I'm not gonna continue, but I'm not telling you why."

If the playername is supposed to already be truncated before getting here, use an ASSERT, if it isn't, then truncate it silently here.